common/rpc/interceptor/telemetry.go

```diff
@@ -37,6 +37,10 @@ import (
 	serviceerrors "go.temporal.io/server/common/serviceerror"
 )
 
+const (
+	ContextKeyMetrics = "context-key-metrics"
+)
+
```

Comment:
There is an encapsulation mehanizm for context:
```suggestion
type (
	metricsContextKey struct{}
)
var (
	metricsCtxKey = metricsContextKey{}
)
```
You will need to move:
```
func (wh *WorkflowHandler) metricsScope(ctx context.Context) metrics.Scope {
```
to this package too but on a bright side it will be real encapsulation.

Reply:
We currently use both string and custom types but I would switch everything to custom types for consistency.